worker/uniter/remotestate/watcher.go

```diff
@@ -251,10 +251,10 @@ func (w *RemoteStateWatcher) loop(unitTag names.UnitTag) (err error) {
 	requiredEvents++
 
 	var (
-		seenStorageChange bool
-		storageChanges    watcher.StringsChannel
-
 		seenApplicationChange *bool
+
+		upgradeSeriesChanges    watcher.NotifyChannel
+		seenUpgradeSeriesChange *bool
```

Comment:
pointer to bool seems so fragile. Do we need the 'nil' state to know that it was never initialized? It seems safer to just always have the bool even if it doesn't mean anything sometimes.
I realize seenApplicationChange follows that pattern, but it just feels like an easy way to mess up and get a panic.

Reply:
I've reworked it